nixos/modules/services/x11/clight.nix

```diff
@@ -0,0 +1,149 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.clight;
+  clightConf = pkgs.writeText "clight.conf" cfg.extraConfig;
+in {
+  options.services.clight = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable clight or not.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.clight;
+      defaultText = "pkgs.clight";
+      description = ''
+        Clight derivation to use.
+      '';
+    };
+
+    clightd = mkOption {
+      type = types.package;
+      default = pkgs.clightd;
+      defaultText = "pkgs.clightd";
+      description = ''
+        Clightd derivation to use.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        Additional configuration to pass to clight.conf.
+      '';
+    };
+
+    extraOptions = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      description = ''
+        Additional command-line arguments to pass to
+        <command>clight</command>.
+      '';
+    };
+
+    latitude = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Your current latitude, between
+        <literal>-90.0</literal> and <literal>90.0</literal>. Must be provided
+        along with longitude.
+      '';
+    };
+
+    longitude = mkOption {
```

Comment:
I feel like it might be a good idea to abstract away longitude and latitude into a `location.nix` module, redshift is another module that requires these.

Reply:
Where should `location.nix` be? From what I'm thinking it would be either `config/` or `misc/`. As for the general namespace I'm thinking about either going under `networking.geolocation` or probably just under `geolocation` with `longitude` and `latitude`.

Reply:
Maybe https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/config/timezone.nix could be repurposed into `locale.nix`. Seems that `location.longitude` would then be a good fit, going along with `time.timeZone`.